workshop/content/advanced/resource-types/example-in_Python/_index.md

```diff
@@ -0,0 +1,302 @@
+
+```shell
+$ git clone https://github.com/aws-cloudformation/aws-cloudformation-samples.git
+```
+
+The repository contains a number of other examples. Change directory into the following one that contains the example resource type, and then choose to enter the directory containing the example implementation in Python:
+
+```shell
+$ cd aws-cloudformation-samples/resource-types/awssamples-ec2-importkeypair/
+$ cd python/
+```
+
+Let's take a look at a number of elements in the directory structure inside of the `python/` directory:
+
+* `README.md`: main documentation file where you'd typically want to add information on how to use a resource type you develop;
+* `docs`: directory containing auto-generated syntax information for properties of the resource type. Every time you make changes to the resource schema file, you want to refresh auto-generated code - that includes files in the `docs/` directory - with the `cfn generate` CloudFormation CLI command;
+* `inputs`: directory containing files with key/value data for resource type input properties. The resource type creator specifies this input information for use with contract tests: *do not add sensitive information to those files*. For more information, see [Specifying input data for use in contract tests](https://docs.aws.amazon.com/cloudformation-cli/latest/userguide/resource-type-test.html#resource-type-test-input-data);
+* `awssamples-ec2-importkeypair.json`: this file, named after the resource type name choice, contains the schema for the resource. When you develop a resource type, you use this file to **describe the model for the resource**;
+* `src`: contains a directory named after the resource type name, inside of which you should find:
+  - the `models.py` file, that is managed by the CloudFormation CLI on your behalf when you make schema changes, and
+  - the `handlers.py` file, that is where the resource type developer adds code for the CRUDL implementation logic. Open the `src/handlers.py` file with a text editor of your choice, and **familiarize with the handlers' structure** described in `create_handler`, `update_handler`, `delete_handler`, `read_handler`, `list_handler` functions;
+* `resource-role.yaml`: file managed by the CloudFormation CLI, that describes an [AWS Identity and Access Management](https://aws.amazon.com/iam/) (IAM) role whose `PolicyDocument` contains permissions the resource type developer indicates in the `handlers` section of the schema file.  CloudFormation assumes this role to manage resources on behalf of the user as part of CRUDL operations;
```

Comment:
Although it is implied, we should explicitly state that the role here is used to manage resources _for this resource type_